### ISSUE TITLE
Ignoring "inline style" rule (H021) for background images

### ISSUE THREAD

**Issue by sveetch** [Riverside-Healthcare/djLint#288]
- [x] I've searched the [issues](https://github.com/Riverside-Healthcare/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## Feature Request

I'm currently starting to adopt djLint to enforce quality on our projects templates, this is a great tool.

Even i endorse the "no inline style" rule (H021), in practice i know that this may be required for background images for some layouts especially in CMS context. So sadly, i've disabled this rule globally, because i found it painful to add a `djlint:off` tag everywhere.

But i'm thinking this may be achieved by modifying the rule regex, does it seems possible to developers ? does someone here have already overrided this regex to not match inline style which only define the `background` or `background-image` property ?

Note: At first i thought it may be cleaner to ask this directly on the discord but it seems your invitation link has expired.

**Comment by christopherpickering** [comment]
Thanks! which invite link was broken? https://discord.gg/taghAqebzU 

For your rule - 

I'd disable H021 as you did, then add a custom rule file `.djlint_rules.yaml`.

```yaml
- rule:
    name: C001
    message: Inline styles should be avoided.
    flags: re.I
    patterns:
      - <\w+\s(?:[^>]*\s)?style=(?=['|\"](?!>|background).*>)
```

This should match any style tags that do not start with `background`. It will allow `background-color`, but you will probably find what you are looking for.

**Comment by sveetch** [comment]
> Thanks! which invite link was broken? https://discord.gg/taghAqebzU
Yep this one, redirection is saying it is not valid anymore although i'm already logged to discord. 

> For your rule -
> 
> I'd disable H021 as you did, then add a custom rule file `.djlint_rules.yaml`.
Yeah, i was pretty sure it was the way.

Thanks for your reply and again for djLint ! Feel free to close this issue.

**Comment by christopherpickering** [comment]
👍🏽 cool! Thank you!

Thanks, I'll get a new link going.
